hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,387 @@
+    api_name = 'http://schema.massopencloud.org/haas/v0/switches/dellnos9'
+
+    __mapper_args__ = {
+        'polymorphic_identity': api_name,
+    }
+
+    id = db.Column(BigIntegerType,
+                   db.ForeignKey('switch.id'), primary_key=True)
+    hostname = db.Column(db.String, nullable=False)
+    username = db.Column(db.String, nullable=False)
+    password = db.Column(db.String, nullable=False)
+    interface_type = db.Column(db.String, nullable=False)
+
+    @staticmethod
+    def validate(kwargs):
+        schema.Schema({
+            'hostname': basestring,
+            'username': basestring,
+            'password': basestring,
+            'interface_type': basestring,
+        }).validate(kwargs)
+
+    def session(self):
+        return self
+
+    @staticmethod
+    def validate_port_name(port):
+        """Valid port names for this switch are of the form 1/0/1 or 1/2"""
+
+        val = re.compile(r'^\d+/\d+(/\d+)?$')
+        if not val.match(port):
+            raise BadArgumentError("Invalid port name. Valid port names for "
+                                   "this switch are of the from 1/0/1 or 1/2")
+        return
+
+    def disconnect(self):
+        pass
+
+    def modify_port(self, port, channel, new_network):
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                self._remove_native_vlan(interface)
+            else:
+                self._set_native_vlan(interface, new_network)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                " switch!"
+            vlan_id = match.groups()[0]
+
+            if new_network is None:
+                self._remove_vlan_from_trunk(interface, vlan_id)
+            else:
+                assert new_network == vlan_id
+                self._add_vlan_to_trunk(interface, vlan_id)
+
+    def revert_port(self, port):
+        """ Removes all vlans"""
+        self._remove_all_vlans_from_trunk(port)
+        if self._get_native_vlan(port) is not None:
+            self._remove_native_vlan(port)
+
+    def get_port_networks(self, ports):
+        """Get port configurations of the switch.
+
+        Args:
+            ports: List of ports to get the configuration for.
+
+        Returns: Dictionary containing the configuration of the form:
+        {
+            Port<"port-3">: [("vlan/native", "23"), ("vlan/52", "52")],
+            Port<"port-7">: [("vlan/23", "23")],
+            Port<"port-8">: [("vlan/native", "52")],
+            ...
+        }
+
+        """
+        response = {}
+        for port in ports:
+            response[port] = filter(None,
+                                    [self._get_native_vlan(port.label)]) \
+                                    + self._get_vlans(port.label)
```

Comment:
This is a bit convoluted; the use of filter on a one-element list seems like overkill; I think a simple if statement would be clearer.

Reply:
I honestly didn't think much about it because I copied this from the brocade too. I'll make it simpler though.